packages/knip/src/plugins/karma/index.ts

```diff
@@ -0,0 +1,57 @@
+import type { IsPluginEnabled, Plugin, ResolveConfig } from '../../types/config.js';
+import { type Input, toDevDependency } from '../../util/input.js';
+import { hasDependency } from '../../util/plugin.js';
+import type { Config, ConfigOptions } from './types.js';
+
+// https://karma-runner.github.io/latest/config/configuration-file.html
+
+const title = 'Karma';
+
+const enablers = ['karma'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+//👇 All but CoffeeScript ones. Low usage nowadays compared to the effort to implement support for those files
+const config = ['karma.conf.js', 'karma.conf.ts', '.config/karma.conf.js', '.config/karma.conf.ts'];
+
+const production: string[] = [];
+
+type ConfigFile = (config: Config) => void;
+/**
+ * 👇 Dummy configuration object that just stores the given config
+ *    Real one merges configurations with Lodash
+ *    https://github.com/karma-runner/karma/blob/v6.4.4/lib/config.js#L275
+ */
+class InMemoryConfig implements Config {
+  config?: ConfigOptions;
+  set(config: ConfigOptions) {
+    this.config = config;
+  }
+}
+
+const resolveConfig: ResolveConfig<ConfigFile> = async configFile => {
+  const inputs = new Set<Input>();
+
+  const inMemoryConfig = new InMemoryConfig();
+  configFile(inMemoryConfig);
+  const { config } = inMemoryConfig;
```

Comment:
A weird way of getting config, which matches the weirdness of the configuration file: a file that exports a function that takes a configuration object which has a `set` method to merge configurations and expects you to call it to set your configuration. So you need to call the exported function with the base configuration where the user will call the `set` method with some configuration options.

Checkout [Karma config file parsing](https://github.com/karma-runner/karma/blob/v6.4.4/lib/config.js#L468-L473)

Reply:
Yeah sometimes it gets a bit weird just to not depend on the actual (often large) tool. Thanks for linking it up in the source, it'll def help in the future.